### PR TITLE
Devel env convenience

### PR DESCRIPTION
**Pull request by erdii**
2 changes:
- gitignore autogenerated cruft
- add editorconfig to automatically set correct indent settings in editorconfig-compatible editors

### CODE REVIEW

**Comment by erdii** [comment]
test failing with

```
16:33:38 FATAL: Vault credentials not found for 'app-sre/quay/none-pull'
16:33:38 com.datapipe.jenkins.vault.exception.VaultPluginException: Vault credentials not found for 'app-sre/quay/none-pull'
16:33:38 	at com.datapipe.jenkins.vault.VaultBuildWrapper.responseHasErrors(VaultBuildWrapper.java:198)
16:33:38 	at com.datapipe.jenkins.vault.VaultBuildWrapper.provideEnvironmentVariablesFromVault(VaultBuildWrapper.java:159)
16:33:38 	at com.datapipe.jenkins.vault.VaultBuildWrapper.setUp(VaultBuildWrapper.java:94)
16:33:38 	at jenkins.tasks.SimpleBuildWrapper.setUp(SimpleBuildWrapper.java:146)
16:33:38 	at hudson.model.Build$BuildExecution.doRun(Build.java:157)
16:33:38 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:504)
16:33:38 	at hudson.model.Run.execute(Run.java:1838)
16:33:38 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
16:33:38 	at hudson.model.ResourceController.execute(ResourceController.java:97)
16:33:38 	at hudson.model.Executor.run(Executor.java:429)
```

**Comment by apahim** [comment]
Are you looking into the CI?

**Comment by erdii** [comment]
Yup

**Comment by erdii** [comment]
CI fixed - @apahim your stage :)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
```
Comment (apahim) [inline .editorconfig:12]: it's funny to see a "No newline at the end of file" message right after a `insert_final_newline = true` setting :)

Comment (erdii) [reply .editorconfig]: hahaha - fixed!

**Review by apahim — APPROVED** [review]
lgtm, tks!

**Pull request merged**